docs/codeql/codeql-for-visual-studio-code/running-codeql-queries-at-scale-with-mrva.rst

```diff
@@ -153,6 +153,32 @@ For example, if you want to continue analyzing a set of repositories that had re
 
 You can then insert the ``new-repo-list`` of repositories into your list of custom repository lists for easy access in the Variant Analysis Repositories panel.
 
+Using code search to add repositories to a custom list 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use code search directly in the CodeQL extension to add a subset of repositories from GitHub.com to a custom list. Note that this feature uses the legacy code search via the code search API. For more information on the syntax to use, see "`Searching code (legacy) <https://docs.github.com/en/search-github/searching-on-github/searching-code>`__."
+
+For example, to add all repositories in the ``rails`` organization on GitHub, you can search ``org:rails``. 
+
+Custom lists can contain a maximum of 1000 repositories, so at most only the first 1000 repositories returned from your search will be added to your list.
+
+#. In the Variant Analysis Repositories panel, choose the list that you want to add repositories to. You can create a new list or choose an existing list that already contains repositories.
+
+#. Right-click on the list you have chosen and then click **Add repositories with GitHub Code Search**. 
+
+#. In the search bar, select a language for your search from the choices in the dropdown. 
+
+    .. image:: ../images/codeql-for-visual-studio-code/variant-analysis-code-search-language.png
+        :alt: Screenshot of the search bar for using code search to add repositories to a custom list. The search bar asks you to choose a language for your search and has a dropdown list of languages to choose from.
+
+#. Type the search query that you want to use and press **Enter**.
+
+#. You can view the progress of your search in the bottom right corner of the application in a box with the text "Searching for repositories...". If you click **Cancel**, no repositories will be added to your list.
+
+#. Once complete, you will see the resulting repositories appear in the dropdown under your custom list in the Variant Analysis Repositories panel.
+
+It is possible that some of the resulting repositories do not have CodeQL databases or are not accessible. When you run an analysis on the list, the results view will show you which repositories could not be analyzed.
```

Comment:
I think it would be helpful here to know how these are reported in the results view.

I suggest that we also reframe this slightly as I suspect that it will be normal to find some repositories without CodeQL databases.

I'm also wondering what would cause a repository to be accessible to your search, but where the CodeQL repository wasn't accessible. 🤔 @norascheuch - can you help with this?

Reply:
Good points! @norascheuch had mentioned in the docs-content issue that access could have been denied to the repository when the analysis is run, but it would be great to get more details if we want to add them to the docs!

I hesitated to go into further detail because from what I could tell, this is not necessarily specific to this feature, it is specific to MRVA generally and the results view of an analysis, which I believe would be covered in this section above: https://codeql.github.com/docs/codeql-for-visual-studio-code/running-codeql-queries-at-scale-with-mrva/#exploring-your-results. We could just link to that section at the end of this paragraph, since it does show a results view, though not specifically repositories that weren't analyzed.

Reply:
In that case, maybe we could say what the results view would report?

Reply:
Updated with a little more info as far as I know (the repositories are broken into three categories in the results view - those that were analyzed, those that denied access, and those that had no CodeQL database

Reply:
Sounds very good, thanks @saritai!

Reply:
> I'm also wondering what would cause a repository to be accessible to your search, but where the CodeQL repository wasn't accessible

Not sure if it's worth going into detail in these docs, but to answer your question I'll try to list all the ways a repo might end up not being analysed.
- If the repo doesn't have a CodeQL database. This is already mentioned. There is a proposal for how we could teach the search API about CodeQL databases, but right now there's no guarantee that repos in the search results have CodeQL databases.
- If the repo is private and you don't have access to it. Though this shouldn't happen much if at all when the results came from the a search, because the search API will have already filtered the list to repos you can see. It's possible if repo visibility changes between running the search and running the MRVA.
- If the repo is private but the controller repo is public. For safety reasons we don't allow analysing private code in the public controller repo. The user could fix this themselves by adding `is:public` to their search. Sadly I don't think we could add this ourselves since at the point of the search we don't know if the controller repo is public or private.

Reply:
Thanks so much for the details! I might hold off on adding them to the docs now, but we can always add more detail later if it turns out to be something we think should be there!